spec/lang/intrinsic.md

```diff
@@ -199,16 +199,19 @@ impl<M: Memory> Machine<M> {
             throw_ub!("invalid first argument to `Intrinsic::Spawn`, function takes arguments");
         }
 
-        // This is taken from Miri. It also does not allow for a return value in the root function of a thread.
-        if func.ret.is_some() {
-            throw_ub!("invalid first argument to `Intrinsic::Spawn`, function returns something");
+        // This is taken from Miri. It discards the return value of the function.
+        // We enforce this by only allowing functions that return () and !.
+        if let Some((name, _)) = func.ret {
+            if !is_unit(func.locals[name].ty) {
```

Comment:
I think we should either force it to be always unit, or just ignore the return type. Or maybe accept all types of size 0. Accepting exactly `!` and `()` is somewhat odd.